sig-architecture/production-readiness.md

```diff
@@ -84,62 +140,26 @@ aspects of the process.
-
-#### Dependencies
-
-* **Does this feature depend on any specific services running in the cluster?**
-  Think about both cluster-level services (e.g. metrics-server) as well
-  as node-level agents (e.g. specific version of CRI). Focus on external or
-  optional services that are needed. For example, if this feature depends on
-  a cloud provider API, or upon an external software-defined storage or network
-  control plane.
-
-* **How does this feature respond to complete failures of the services on which
-  it depends?**
-  Think about both running and newly created user workloads as well as
-  cluster-level services (e.g. DNS).
-
-* **How does this feature respond to degraded performance or high error rates
-  from services on which it depends?**
-
-#### Monitoring requirements
-
-* **How can an operator determine if the feature is in use by workloads?**
-
-* **How can an operator determine if the feature is functioning properly?**
-  Focus on metrics that cluster operators may gather from different
-  components and treat other signals as last resort.
-
-* **What are the SLIs (Service Level Indicators) an operator can use to
-  determine the health of the service?**
-  - [ ] Metrics
-    - Metric name:
-    - [Optional] Aggregation method:
-    - Components exposing the metric:
-  - [ ] Other (treat as last resort)
-    - Details:
-
-* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
 
 #### Troubleshooting
 Troubleshooting section serves the `Playbook` role as of now. We may consider
 splitting it into a dedicated `Playbook` document (potentially with some monitoring
 details). For now we leave it here though, with some questions not required until
 further stages (e.g. Beta/Ga) of feature lifecycle.
 
-* **How does this feature react if the API server is unavailable?**
+* **How does this feature react if the API server and/or etcd is unavailable?**
 
 * **What are other known failure modes?**
-
-* **How can those be detected via metrics or logs?**
-  Stated another way: how can an operator troubleshoot without logging into a
-  master or worker node?
-
-* **What are the mitigations for each of those failure modes?**
-
-* **What are the most useful log messages and what logging levels to they require?**
-  Not required until feature graduates to Beta.
+  For each of them fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without loogging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+	  - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debugging the issue?
+      Not required until feature graduated to Beta.
+    - Testing: Are there any tests for failure mode? If not describe why.
```

Comment:
to we have the ability to tests most failure modes now?  I don't recall disruptive or failure mode tests for kube in general because many of them require knowledge of how kube was deployed (openshift's restore etcd from backup, tests as a for instance)

Reply:
We don't have generic tests that prove it generally works. But we have a number of disruption tests working for *some* setups/cloud providers. I agree that it's not perfect coverage, but it's at least some coverage (at least for generic non-provider oriented parts of the feature).